application/CMakeLists.txt

```diff
@@ -56,11 +56,20 @@ elseif (MSVC)
   set_target_properties(f3d PROPERTIES VS_DEBUGGER_ENVIRONMENT "PATH=$<TARGET_FILE_DIR:VTK::CommonCore>") # Add VTK in MSVC PATH environment variable
 endif ()
 
-target_include_directories(f3d
-  PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external>
+if (F3D_USE_EXTERNAL_DEPS)
+  find_package(cxxopts REQUIRED)
+  find_package(nlohmann_json REQUIRED)
+  target_include_directories(f3d
```

Comment:
I think you need `target_link_libraries(f3d PRIVATE cxxopts::cxxopts)` and `target_link_libraries(f3d PRIVATE nlohmann_json::nlohmann_json)` to add the include directories of these dependencies in case the install directory is not standard.  
I'm not sure about the target names, so please double check.

Reply:
arent these libs headeronly ?

Reply:
Indeed, but the targets expose the include directories via `INCLUDE_DIRECTORIES` target property set up by the target config file